### DISCUSSION TITLE
Spectating when mapping

### DISCUSSION THREAD

**Discussion by minisbett** [ppy/osu#20126] [Ideas] [answered]
This is just a shower thought and I can't really estimate how complex implementing this idea would be because I don't know the future server <-> client infrastructure but I think it would be really cool to be able to spectate people while mapping so that you can see what they are doing right now, syncing what you currently have selected, what you clicked on, moving circles, scrolling through the timeline etc., even allowing to access stuff like the timing panel while the mapper is mapping. My only concern is that the rate at which updates would be pushed to the clients through the server in order to have a good refresh rate of events happening might be too high for the server to handle properly and if so, the idea wouldn't work that great. It's just an idea, after all. And ofcourse, it should be toggleable by mappers so that they can also map without being disturbed. I think that it would be cool to be able to watch your favorite mapper mapping without them having to stream somewhere because maybe they don't have the right setup or knowledge to set everything up. You would also have the ability to communicate with spectators using the chat to get opinions and give advice, all in one I think it would give some great touch to the connection between not-into-mapping players and the mapping process.

**Comment by Theighlin** [comment, marked as answer]
Maybe a duplicate of #14556 and #13046. This is those but with the mapping together feature removed. I imagine the work to be done code-wise is almost the same.
